File: mediator/exercise/AirportControlTower.java
package mediator.exercise;

import java.util.ArrayList;
import java.util.List;

public class AirportControlTower implements AirTrafficControlMediator {

    private List<Airplane> airplanes;

    public AirportControlTower() {
        this.airplanes = new ArrayList<>();
    }

    @Override
    public void requestTakeOff(Airplane airplane) {
        airplanes.forEach(otherAirplanes -> otherAirplanes.notifyAirTrafficControl("the following airplane: "
                + airplane.getType() + " is taking off..."));
        airplane.notifyAirTrafficControl("Requesting takeoff accepted...");
    }

    @Override
    public void requestLanding(Airplane airplane) {
        airplanes.forEach(otherAirplanes -> otherAirplanes.notifyAirTrafficControl("the following airplane: "
                + airplane.getType() + " is landing..."));
        airplane.notifyAirTrafficControl("Requesting landing accepted...");
    }

    @Override
    public void register(Airplane airplane) {
        airplanes.add(airplane);
    }

}

File: C#/Mediator/Exercise/AirportControlTower.cs
using System.Runtime.InteropServices;

namespace Mediator.Exercise;

public sealed class AirportControlTower : IAirTrafficControlMediator
{
    private readonly List<IAirplane> _airplanes;

    public AirportControlTower()
        => _airplanes = [];

    public void Register(IAirplane airplane)
        => _airplanes.Add(airplane);

    public void RequestLanding(IAirplane airplane)
    {
        foreach (var otherAirplane in CollectionsMarshal.AsSpan(_airplanes))
        {
            otherAirplane.NotifyAirTrafficControl($"the following airplane: {airplane.Type} is taking off...");
        }

        airplane.NotifyAirTrafficControl("Requesting takeoff accepted...");
    }

    public void RequestTakeOff(IAirplane airplane)
    {
        foreach (var otherAirplane in CollectionsMarshal.AsSpan(_airplanes))
        {
            otherAirplane.NotifyAirTrafficControl($"the following airplane: {airplane.Type} is landing...");
        }

        airplane.NotifyAirTrafficControl("Requesting landing accepted...");
    }
}
